presto-hive/src/test/java/com/facebook/presto/hive/TestSymlinkTableListCaching.java

```diff
@@ -0,0 +1,136 @@
+import org.testng.annotations.Test;
+
+import java.util.Optional;
+import java.util.function.Consumer;
+
+import static com.facebook.presto.hive.HiveQueryRunner.METASTORE_CONTEXT;
+import static com.facebook.presto.hive.HiveQueryRunner.getFileHiveMetastore;
+import static org.testng.Assert.assertEquals;
+
+public class TestSymlinkTableListCaching
+        extends AbstractTestQueryFramework
+{
+    ExtendedHiveMetastore metastore;
+    private static final PrincipalPrivileges PRINCIPAL_PRIVILEGES = new PrincipalPrivileges(ImmutableMultimap.of(), ImmutableMultimap.of());
+    @Override
+    protected QueryRunner createQueryRunner() throws Exception
+    {
+        ImmutableMap.Builder<String, String> extraHiveProperties = ImmutableMap.<String, String>builder()
+                .put("hive.file-status-cache-expire-time", "24h")
+                .put("hive.file-status-cache-size", "100000000")
+                .put("hive.file-status-cache-tables", "*");
+
+        DistributedQueryRunner queryRunner = HiveQueryRunner.createQueryRunner(
+                ImmutableList.of(),
+                ImmutableList.of(),
+                ImmutableMap.of(),
+                ImmutableMap.of(),
+                "sql-standard",
+                extraHiveProperties.build(),
+                Optional.of(1),
+                Optional.empty(),
+                Optional.empty(),
+                Optional.empty(),
+                true);
+
+        // setup file metastore for creating table
+        metastore = getFileHiveMetastore(queryRunner);
+        return queryRunner;
+    }
+
+    @BeforeClass
+    public void setUp()
+    {
+        assertQuerySucceeds("CREATE SCHEMA hive.test_dbname");
+        createSymlinkTable("hive_symlink_table");
+    }
+
+    @Test
+    public void testSymlinkTableListCaching()
+    {
+        String jmxMetricsQuery = "SELECT hitcount, misscount from jmx.current.\"com.facebook.presto.hive:name=hive,type=cachingdirectorylister\"";
+        String symlinkTableQuery = "SELECT count(*) FROM hive.test_dbname.hive_symlink_table";
+
+        // Initial cache entries, hitcount, misscount will all be zero
+        assertQuery(jmxMetricsQuery, "VALUES (0, 0)");
+
+        for (int i = 0; i < 2; i++) {
+            assertQuerySucceeds(symlinkTableQuery);
+        }
+
+        MaterializedResult result = computeActual(jmxMetricsQuery);
+
+        assertEquals(result.getRowCount(), 1);
+
+        long hitCount = (long) result.getMaterializedRows().get(0).getField(0);
+        long missCount = (long) result.getMaterializedRows().get(0).getField(1);
+
+        assertEquals(hitCount, 1);
+        assertEquals(missCount, 1);
+    }
+
+    @AfterClass(alwaysRun = true)
+    public void tearDown()
+    {
+        assertQuerySucceeds("DROP TABLE IF EXISTS hive.test_dbname.hive_symlink_table");
+        assertQuerySucceeds("DROP SCHEMA IF EXISTS  hive.test_dbname");
+    }
+
+    protected static String goldenTablePath(String tableName)
```

Comment:
nit: can we update the method name to be more intuitive? or add a comment to define what is goldenTablePath